src/tiled-map-resource.ts

```diff
@@ -231,49 +253,53 @@ export class TiledMapResource implements Loadable<TiledMap> {
 
    private _parseExcaliburInfo() {
       // Tiled+Excalibur smarts
-      const excalibur = this.data?.getExcaliburObjects();
+      const excaliburObjectLayers = this.data?.getExcaliburObjects();
 
       const ex: ExcaliburData = {};
-      if (excalibur.length > 0) {
+      if (excaliburObjectLayers.length > 0) {
          // Parse cameras find the first
-         ex.camera = excalibur.find(objectlayer => objectlayer.getObjectByType('camera'))?.getCamera();
+         ex.camera = excaliburObjectLayers.find(objectlayer => objectlayer.getObjectByType('camera'))?.getCamera();
          // Parse colliders
          ex.colliders = [];
-         const boxColliders = excalibur.flatMap(o => o.getObjectsByType('boxcollider'));
-         for (let box of boxColliders) {
-            const collisionType = box.getProperty<CollisionType>('collisiontype');
-            const color = box.getProperty<string>('color');
-            const zIndex = box.getProperty<number>('zindex');
-            ex.colliders.push({
-               ...box,
-               width: +(box.width ?? 0),
-               height: +(box.height ?? 0),
-               collisionType: collisionType?.value ?? CollisionType.Fixed,
-               color,
-               zIndex: +(zIndex?.value ?? 0),
-               radius: 0,
-               type: 'box',
-               tiled: box
-            });
-         }
+         for (let objectLayer of excaliburObjectLayers) {
+
+            const boxColliders = objectLayer.getObjectsByType('boxcollider');
+
+            for (let box of boxColliders) {
+               const collisionType = box.getProperty<CollisionType>('collisiontype');
+               const color = box.getProperty<string>('color');
+               const zIndex = box.getProperty<number>('zindex')?.value ?? (objectLayer.rawObjectGroup.order + this._layerZIndexStart);
+               ex.colliders.push({
+                  ...box,
+                  width: +(box.width ?? 0),
+                  height: +(box.height ?? 0),
+                  collisionType: collisionType?.value ?? CollisionType.Fixed,
+                  color,
+                  zIndex: zIndex,
```

Comment:
Does this need to be coerced to int? When a custom value exists, it gets passed raw it looks like. Speaks to extracting a helper to handle getting this value vs. passing raw from `getProperty` otherwise it's easy to miss.

Reply:
I'll make the helper coerce the type, but it's definitely good to do 👍